### ISSUE TITLE
RectangularRadioButton problems related to appearance strategies

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/sun#772]
A couple of TypeScript problems in RectangularRadioButton.ts:

* FlatAppearanceStrategyOptions and ContentAppearanceStrategyOptions are missing, and `any` is used.

* SelfOptions duplicates a bunch of definitons that should be in FlatAppearanceStrategyOptions, ContentAppearanceStrategyOptions

* These comment in SelfOptions are troubling, I don't know how someone would "add your own options".  And these should be nested options.

```js
  // Options used by RectangularRadioButton.FlatAppearanceStrategy.
  // If you define your own buttonAppearanceStrategy, then you may need to add your own options.
...
  // Options used by RectangularRadioButton.ContentAppearanceStrategy.
  // If you define your own contentAppearanceStrategy, then you may need to add your own options.
```

Assigning to @jbphet since he's the architect of appearance strategies.

**Comment by pixelzoom** [comment]
Note that those options that should be in FlatAppearanceStrategyOptions and  ContentAppearanceStrategyOptions are also repeated in RectangularRadioButtonGroup.  Discovered in https://github.com/phetsims/sun/issues/740.  So this will also need to be addressed in RectangularRadioButtonGroup.

**Comment by pixelzoom** [comment]
Adding block-publication to this issue because radio button appearance is currently broken.  Stroke is missing, not sure if there are other problems.    Some examples...

Radio buttons in EqEx missing strokes:

![image](https://user-images.githubusercontent.com/3046552/180327532-0d857f40-d4c1-4476-b187-c02398377795.png)

MOTHA unusable without radio button strokes:

![image](https://user-images.githubusercontent.com/3046552/180327628-4002db93-22a8-411f-af30-7f0f38929997.png)

**Comment by jessegreenberg** [comment]
After pulling today I also noticed that button strokes have gotten a little thicker
Master:
![image](https://user-images.githubusercontent.com/6396244/180477445-e06ea674-3e7b-4b8b-a8fe-200aee93f992.png)

Published:
![image](https://user-images.githubusercontent.com/6396244/180477472-190d4594-9b23-4e15-bdd1-693a0563252f.png)

And rectangular radio buttons might be missing an opacity or filter on deselected buttons:
Master:
![image](https://user-images.githubusercontent.com/6396244/180479042-7d8e8097-5b02-4aa4-9d7f-3576a31b4564.png)

Published:
![image](https://user-images.githubusercontent.com/6396244/180479008-da6c4bbf-4210-45c0-8f14-e5ed198e8690.png)

**Comment by pixelzoom** [comment]
Raising the priority of this.  This has been broken for way too long.  @jbphet can you please take a look?

**Comment by pixelzoom** [comment]
@Nancy-Salpepi reported this problem in Slack#dev-public, which seems related:

> Hi. I was testing Gravity and Orbits and I noticed that when I hover over different scenes an outline now appears around them that isn’t present in the published version or older dev versions. Was this an intentional change? Here is an example with the earth/moon scene on projector mode:

<img width="283" alt="Screen Shot 2022-09-27 at 3 10 51 PM" src="https://user-images.githubusercontent.com/3046552/192617318-eee853c4-f838-4618-a021-506cde88af8c.png">

**Comment by jbphet** [comment]
I've investigated this, and found that some of the changes I made when porting this to TypeScript removed some default behaviors that were needed.  I've put those back, and verified the behavior on several sims (see list below), and they seem to be behaving correctly now.

For reference, the sims I checked were molecules-and-light, equality-explorer, gravity-and-orbits, models-of-the-hydrogen-atom (which isn't published, so I just validated that it looked reasonable), number-line-integers, area-model-algebra, and bending-light.

I'd like to review the current code with at least one other developer.  The appearance strategy uses the values for the `deselected` case for the `over` case in a number of places, which strikes me as a little odd, but many of the sims are relying on this behavior, so perhaps we just leave it.
